test/operand_pattern_test.cpp

```diff
@@ -199,20 +199,29 @@ TEST(AlternatePatternFollowingImmediate, MultipleNonResultIds) {
 TEST(AlternatePatternFollowingImmediate, ResultIdFront) {
   EXPECT_THAT(spvAlternatePatternFollowingImmediate(
                   {SPV_OPERAND_TYPE_RESULT_ID, SPV_OPERAND_TYPE_ID}),
-              Eq(spv_operand_pattern_t{SPV_OPERAND_TYPE_RESULT_ID,
+              Eq(spv_operand_pattern_t{SPV_OPERAND_TYPE_OPTIONAL_CIV,
```

Comment:
Are the input parameters not supposed to be in the reversed order?
```
EXPECT_THAT(spvAlternatePatternFollowingImmediate(
    {SPV_OPERAND_TYPE_ID, SPV_OPERAND_TYPE_RESULT_ID}),
```
I don't know if it's important, but it looks like you are changing the original tests if the input parameters are given in the same order as before, but the function expects them to be reversed.